website/parser_test.ts

```diff
@@ -0,0 +1,458 @@
+  assertEqual(D06.extendsType.name, "Y");
+  // Test falseType
+  assertEqual(D06.falseType.type, "keyword");
+  assertEqual(D06.falseType.name, "string");
+  // Test trueType
+  assertEqual(D06.trueType.type, "keyword");
+  assertEqual(D06.trueType.name, "number");
+  // Skipping parameters, tested several times.
+
+  // Test T07
+  const T07 = docs[7];
+  assertEqual(T07.name, "T07");
+  assertEqual(T07.documentation, "comment 10");
+  // Test definition
+  const D07 = T07.definition;
+  assertEqual(D07.type, "MappedType");
+  // No need to test D07.parameter completely, just check if it contains that.
+  assertEqual(D07.parameter.type, "TypeParameter");
+  assertEqual(D07.parameter.name, "P");
+  // IndexedAccessTypeNode has been already tested
+  assertEqual(D07.dataType.type, "IndexedAccessTypeNode");
+
+  // Test T08
+  const T08 = docs[8];
+  assertEqual(T08.name, "T08");
+  assertEqual(T08.documentation, "comment 11");
+  assertEqual(T08.definition.dataType.type, "UnionType");
+  assertEqual(T08.definition.dataType.types.length, 2);
+  assertEqual(T08.definition.dataType.types[1].type, "keyword");
+  assertEqual(T08.definition.dataType.types[1].name, "null");
+});
+
+test(async function test_findDefinition() {
+  const doc = generateDoc("testdata/import.ts", options);
+  const Y = doc[1];
+  assertEqual(Y.name, "Y");
+  const YF = Y.statements[0];
+  assertEqual(YF.name, "F");
+  assertEqual(YF.parameters[0].dataType.filename, "http://site.com/foo");
+  const YP = Y.statements[1];
+  assertEqual(YP.name, "P");
+  const YPT = YP.statements[0];
+  assertEqual(YPT.name, "T");
+  assertEqual(YPT.parameters[0].dataType.filename, "#Y.P");
+  const YPG = YP.statements[1];
+  assertEqual(YPG.name, "G");
+  assertEqual(YPG.parameters[0].dataType.filename, "#Y.P");
+  const YPF = YP.statements[2];
+  assertEqual(YPF.name, "F");
+  assertEqual(YPF.parameters[0].dataType.filename, "http://site.com/foo");
+  const YG = Y.statements[2];
+  assertEqual(YG.name, "G");
+  assertEqual(YG.parameters[0].dataType.filename, "http://site.com/bar");
+});
+
+test(async function test_export() {
+  const doc = generateDoc("testdata/export.ts", options);
+  const Test = doc.filter(x => x.name === "Test")[0];
+  assertEqual(!!Test, true);
+  assertEqual(Test.isPrivate, true);
+  const X = doc.filter(
+    x => x.type === "VariableDeclarationList" && x.declarations[0].name === "X"
+  );
+  assertEqual(X.length, 1);
+  assertEqual(X[0].declarations[0].name, "X");
+  assertEqual(X[0].isPrivate, true);
+  const P = doc.filter(
+    p => p.type === "VariableDeclarationList" && p.declarations[0].name === "P"
+  );
+  assertEqual(P.length, 1);
+  assertEqual(P[0].declarations[0].name, "P");
+  assertEqual(P[0].isPrivate, true);
+});
+
+test(async function test_js() {
+  const doc = generateDoc("testdata/file.js", options);
+  assertEqual(doc.length, 4);
+  const fooExport = doc.filter(x => x.name === "foo" && x.type === "export");
```

Comment:
When I print "fooExport" I get
```
[ { type: 'export',
    propertyName: 'foo',
    name: 'foo',
    filename: '#' } ]
```
There is no way to see "Some comments" here? Why is filename "#"?

Reply:
Can you add a test that tests it?

Reply:
ok

Reply:
@ry it's in `fooFn` i just need to test its documentation string

Reply:
done

Reply:
But we're really interested in the documentation of the exports of a file. Given the exports, what is the documentation?